internal/scorecard/tests/olm.go

```diff
@@ -219,8 +219,44 @@ func checkCSVDescriptors(bundle *apimanifests.Bundle, r scapiv1alpha3.TestResult
 	}
 	r.Log += fmt.Sprintf("Loaded %d Custom Resources from alm-examples\n", len(crs))
 
+	// if the descriptor is status then check the status
+	// otherwise check the spec
 	for _, cr := range crs {
-		r = checkOwnedCSVDescriptors(cr, bundle.CSV, descriptor, r)
+		if descriptor == statusDescriptor {
+			r = checkOwnedCSVStatusDescriptor(cr, bundle.CSV, r)
+		} else {
+			r = checkOwnedCSVSpecDescriptors(cr, bundle.CSV, r)
+		}
+	}
+	return r
+}
+
+func checkOwnedCSVStatusDescriptor(cr unstructured.Unstructured, csv *operatorsv1alpha1.ClusterServiceVersion,
+	r scapiv1alpha3.TestResult) scapiv1alpha3.TestResult {
+
+	var crd *operatorsv1alpha1.CRDDescription
+
+	if cr.Object[statusDescriptor] == nil {
+		r.Suggestions = append(r.Suggestions, fmt.Sprintf("Status field can be added to the CR %s", cr.GetName()))
```

Comment:
I thought the whole point of this PR was to remove this check entirely @varshaprasad96?

Reply:
Since having `Status` field is suggested according the k8s-api conventions, have modified this check to add suggestions. Also [this unit test](https://github.com/operator-framework/operator-sdk/pull/4009/files#diff-e41b6bcfe63a0c48f173be8249e4ba79879e91bf1048554e1470f64679b37315R179) ensures that if `Object` descriptor is nil, the scorecard test still passes.

Reply:
Instead of [failing](https://github.com/operator-framework/operator-sdk/blob/9d27e224efac78fcc9354ece4e43a50eb30ea968/internal/scorecard/tests/olm.go#L235-L238) when the descriptor is not present, suggestions are added. And we directly proceed with checking if the `statusDescriptor` is present in CRD or not. If the users input as described in #3999 is given with no `status` field in CR, this test would pass.

Reply:
A CR from `alm-examples` should never have a status according to https://github.com/operator-framework/operator-sdk/issues/3999#issuecomment-705718844, so we shouldn't be suggesting that one be added. Whether the CRD defines a `status` field for a resource in its validation is another story.

Reply:
Hmm..okay. Because based on the comment, I tried creating a resource through console UI - with and without `status` field. And it worked in both the cases. And to verify if such CSVs exist, just randomly went through the example community operators and found [this](https://github.com/operator-framework/community-operators/blob/master/community-operators/codeready-toolchain-operator/0.1.3/codeready-toolchain-operator.clusterserviceversion.yaml#L18-L29).

But considering we remove that validation check for that, can we remove the suggestion, or do you suggest we remove the entire `[StatusDescriptor]`(https://github.com/operator-framework/operator-sdk/blob/9d27e224efac78fcc9354ece4e43a50eb30ea968/internal/scorecard/tests/olm.go#L232-L284) validation test?

Shouldn't we still check if `statusDescriptor` is present in CRD?

Reply:
I shouldn't have said "never" above. The `alm-examples` can have `status` blocks, but they should be treated as not existing. Therefore I'm only suggesting the `cr.Object[statusDescriptor] == nil` condition be removed. Everything else is fine.

Reply:
Got it. Will create a follow up PR for this.